pkg/controller/secrets/tenant_controller.go

```diff
@@ -90,6 +90,16 @@ func AddTenantController(mgr manager.Manager, opts options.AddOptions) error {
 		return fmt.Errorf("tenant-controller failed to watch tenant CA Secret %s in all namespace: %w", certificatemanagement.TenantCASecretName, err)
 	}
 
+	// Make a helper for determining which namespaces to use based on tenancy mode.
+	// In multi-tenant mode, we need to watch all namespaces for secrets.
+	helper := utils.NewNamespaceHelper(opts.MultiTenant, "", "")
```

Comment:
Maybe just remove the Namespace helper entirely and use `""` in the watches directly since the helper has no effect?